### PR TITLE
feat: ws client docs

### PR DESCRIPTION
**Pull request by agraebe**
Adding docs for JS WebSocket client: https://blockstack.github.io/stacks-blockchain-api/#section/Websockets

### CODE REVIEW

**Comment by agraebe** [comment]
@zone117x could you also update the NPM package to capture the latest README details once merged?

**Comment by agraebe** [comment]
I just reviewed the outcome of the generated docs and figured this requires more work:
- [ ] Update client README to capture that the client include REST API and WebSocket support
- [ ] Review package name inconsistencies (stacks vs @blockstack, client vs ws-client?)
- [ ] Update OpenAPI description to reflect the scope and feature of the package

File: client/README.md [hunk 1]
```diff
@@ -2,8 +2,47 @@
...
+```shell
+npm install --save-dev @blockstack/ws-rpc-client
+```
+
+## Usage
+
+Here is an example code that connects with the WebSocket server and subscribes to updates for a specific Stacks address:
+
+```js
+import { connect as connectWebSocketClient } from '@blockstack/ws-rpc-client';
```
Comment (zone117x) [inline client/README.md:24]: Should be 
```js
import { connectWebSocketClient } from '@stacks/blockchain-api-client';
```

File: client/README.md [hunk 2]
```diff
@@ -2,8 +2,47 @@
 [![NPM Package](https://img.shields.io/npm/v/@stacks/blockchain-api-client.svg?style=flat-square)](https://www.npmjs.org/package/@stacks/blockchain-api-client)
 [![Build Status](https://github.com/blockstack/stacks-blockchain-api/workflows/Build/badge.svg)](https://github.com/blockstack/stacks-blockchain-api/actions)
 
-Client for the Stacks Blockchain API.
+A JS Client for the Stacks Blockchain API. This includes a JSON RPC WebSocket client to subscribe to real-time updates, with full type safety.
 
-### Known Issues
+![image](https://user-images.githubusercontent.com/1447546/89547223-b8aa0980-d7c2-11ea-9aea-658a9dc96a67.png)
```
Comment (zone117x) [inline client/README.md:7]: The code doesn't match these screenshots anymore

**Comment by zone117x** [comment]
@agraebe 
> Review package name inconsistencies (stacks vs @blockstack, client vs ws-client?)

Note we have two packages:
* [`@stacks/blockchain-api-client `](https://github.com/blockstack/stacks-blockchain-api/blob/master/client/package.json#L2) -- the generated client code for the REST and websocket APIs.
* [`@blockstack/stacks-blockchain-api-types`](https://github.com/blockstack/stacks-blockchain-api/blob/master/docs/package.json#L2) -- a typescript definition only package

File: client/README.md [hunk 3]
```diff
@@ -2,8 +2,47 @@
 [![NPM Package](https://img.shields.io/npm/v/@stacks/blockchain-api-client.svg?style=flat-square)](https://www.npmjs.org/package/@stacks/blockchain-api-client)
 [![Build Status](https://github.com/blockstack/stacks-blockchain-api/workflows/Build/badge.svg)](https://github.com/blockstack/stacks-blockchain-api/actions)
```
Comment (zone117x) [inline client/README.md:3]: I left this broken image here initially. If you want to fix, it should be
```
https://github.com/blockstack/stacks-blockchain-api/workflows/stacks-blockchain-api/badge.svg
```

File: client/package.json [hunk 4]
```diff
@@ -26,7 +26,8 @@
     "lint:prettier": "prettier --check ./src/**/*.{ts}",
     "open": "http-server -o 9222 -o index.html",
     "prep-openapi": "rimraf ./.tmp && rimraf ./src/generated && swagger-cli bundle --dereference -o ./.tmp/openapi.json ../docs/openapi.yaml && shx sed -i '^.*\\$schema.*$' '' ./.tmp/openapi.json",
-    "generate-openapi": "npm run prep-openapi && openapi-generator generate --skip-validate-spec -g typescript-fetch --additional-properties=typescriptThreePlus=true,supportsES6=true,legacyDiscriminatorBehavior=false,enumPropertyNaming=original,modelPropertyNaming=original -i ./.tmp/openapi.json -o ./src/generated"
+    "generate-openapi": "npm run prep-openapi && openapi-generator generate --skip-validate-spec -g typescript-fetch --additional-properties=typescriptThreePlus=true,supportsES6=true,legacyDiscriminatorBehavior=false,enumPropertyNaming=original,modelPropertyNaming=original -i ./.tmp/openapi.json -o ./src/generated",
+    "generate:docs": "typedoc"
```
Comment (zone117x) [inline client/package.json:30]: Now that we have 2 commands to run for generating doc html (this one and the one in `/docs`), I think we'll need a script in the top-level package.json that calls both of these before the gulp gh-pages publish action?

Comment (agraebe) [reply client/package.json:30]: Agreed, adding that as well!

File: client/README.md (hunk 3 above)
Comment (agraebe) [reply client/README.md:3]: 👍

File: client/README.md (hunk 2 above)
Comment (agraebe) [reply client/README.md:7]: I added a "features" section to replace what I wanted to communicate through the screenshots

File: package.json [hunk 5]
```diff
@@ -19,7 +19,9 @@
     "devenv:build": "docker-compose -f docker-compose.dev.postgres.yml -f docker-compose.dev.stacks-blockchain.yml -f docker-compose.dev.bitcoind.yml build --no-cache",
     "devenv:deploy": "docker-compose -f docker-compose.dev.postgres.yml -f docker-compose.dev.stacks-blockchain.yml -f docker-compose.dev.bitcoind.yml up",
     "devenv:stop": "docker-compose -f docker-compose.dev.postgres.yml -f docker-compose.dev.stacks-blockchain.yml -f docker-compose.dev.bitcoind.yml down -v -t 0",
-    "devenv:logs": "docker-compose -f docker-compose.dev.postgres.yml -f docker-compose.dev.stacks-blockchain.yml -f docker-compose.dev.bitcoind.yml logs -t -f"
+    "devenv:logs": "docker-compose -f docker-compose.dev.postgres.yml -f docker-compose.dev.stacks-blockchain.yml -f docker-compose.dev.bitcoind.yml logs -t -f",
+    "docs:generate": "cd docs && npm run generate:types; npm run generate:docs; cd ../client && typedoc",
```
Comment (zone117x) [inline package.json:23]: `;` operator not Windows friendly :)

Comment (agraebe) [reply package.json]: thanks!

File: package.json [hunk 6]
```diff
@@ -8,7 +8,7 @@
     "dev:integrated": "npm run devenv:build && concurrently npm:dev npm:devenv:deploy",
     "test": "cross-env NODE_ENV=development jest --config ./jest.config.js --coverage --runInBand",
     "test:watch": "cross-env NODE_ENV=development jest --config ./jest.config.js --watch",
-    "test:integration": "npm run devenv:deploy -- -d && cross-env NODE_ENV=development jest --config ./jest.config.js --coverage --no-cache --runInBand; npm run devenv:stop",
+    "test:integration": "npm run devenv:deploy -- -d && cross-env NODE_ENV=development jest --config ./jest.config.js --coverage --no-cache --runInBand && npm run devenv:stop",
```
Comment (zone117x) [inline package.json:11]: This one has to be `;` to function properly on *nix (cleanup should happen regardless of test exit code). There's probably some npx script to make it cross platform but too much work for now lol

**Pull request merged**